src/tests/ftest/nvme/nvme_health.py

```diff
@@ -0,0 +1,125 @@
+        """
+        Test Description: Test Health monitor for large number of pools.
+        Use Case: This tests smd's following functions: nvme_list_streams,
+                  nvme_get_pool, nvme_set_pool_info, nvme_add_pool,
+                  nvme_get_device, nvme_set_device_status,
+                  nvme_add_stream_bond, nvme_get_stream_bond
+
+        :avocado: tags=all,hw,medium,nvme,ib2,full_regression
+        :avocado: tags=nvme_health,nvme_health_large_pools
+        """
+        # pylint: disable=attribute-defined-outside-init
+        # pylint: disable=too-many-branches
+        no_of_pools = self.params.get("number_of_pools", '/run/pool/*')
+        #Stop the servers to run SPDK too to get the server capacity
+        self.stop_servers_noreset()
+        storage = self.get_server_capacity()
+        self.start_servers()
+
+        single_pool_nvme_size = int((storage * 0.80)/no_of_pools)
+        pools = []
+        #Create the Large number of pools
+        for _pool in range(no_of_pools):
+            pool = TestPool(self.context, dmg_command=self.get_dmg_command())
+            pool.get_params(self)
+            #SCM size is 10% of NVMe
+            pool.scm_size.update('{}GB'.format(single_pool_nvme_size * 0.10))
+            pool.nvme_size.update('{}GB'.format(single_pool_nvme_size))
+            pool.create()
+            pools.append(pool)
+
+        #initialize the dmg command
+        self.dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        self.dmg.get_params(self)
+        self.dmg.insecure.update(
+            self.server_managers[0].get_config_value("allow_insecure"),
+            "dmg.insecure")
+
+        #Get Pool query for SMD
+        self.dmg.set_sub_command("storage")
+        self.dmg.sub_command_class.set_sub_command("query")
+        self.dmg.sub_command_class.sub_command_class.set_sub_command("smd")
+        for host in self.hostlist_servers:
+            self.dmg.sub_command_class. \
+                sub_command_class.sub_command_class.devices.value = False
+            self.dmg.sub_command_class. \
+                sub_command_class.sub_command_class.pools.value = True
+            self.dmg.hostlist = host
+            try:
+                result = self.dmg.run()
+            except CommandFailure as error:
+                self.fail("dmg command failed: {}".format(error))
+            #Verify all pools UUID listed as part of query
+            for pool in pools:
+                if pool.uuid.lower() not in result.stdout:
+                    self.fail('Pool uuid {} not found in smd query'
+                              .format(pool.uuid.lower()))
+
+        # Get the device ID
+        device_ids = get_device_ids(self.dmg, self.hostlist_servers)
+
+        #Get the device states
+        for host in device_ids:
+            self.dmg.hostlist = host
+            for _dev in device_ids[host]:
+                try:
+                    result = self.dmg.storage_query_device_state(_dev)
```

Comment:
(pylint-no-member) Instance of 'DmgCommand' has no 'storage_query_device_state' member